src/main/java/com/gregtechceu/gtceu/api/machine/MetaMachine.java

```diff
@@ -685,15 +690,27 @@ public int getOutputSignal(@Nullable Direction side) {
 
     @Override
     public boolean canConnectRedstone(Direction side) {
-        if (side == null) return false;
```

Comment:
If side is guaranteed to be not null, put the annotation there. I believe this code dates pretty far back and the check exists because either MC called it with null in some scenarios, or mods like TOP or WAILA did.